Added parity transformation for issue #4881 (#4928)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

  ```python
  import pennylane as qml

>   dev = qml.device("default.clifford", tableau=True)  

Reviewer: Why this is changed?

Author: This was a suggested change by @trbromley. I just committed the suggestion there.

  import pennylane as qml

  dev = qml.device("default.clifford", tableau=True)

>   

Reviewer 2: ```suggestion

```

  + 2j * X(0) @ Z(1)
  ```

<h4>Parity Mapping</h4>

> * `parity_transform`  is added for parity mapping of a fermionic Hamiltonian.

Reviewer: ```suggestion
* `parity_transform` is added for parity mapping of a fermionic Hamiltonian.
```

   [(#4928)](https://github.com/PennyLaneAI/pennylane/pull/4928)

>    It is now possible to transform fermionic Hamiltonian to qubit Hamiltonian through parity mapping.

Reviewer: ```suggestion
   It is now possible to transform a fermionic Hamiltonian to a qubit Hamiltonian with parity mapping.
```

   ```python
   import pennylane as qml

>    fermi_ham = qml.FermiWord({(0, 0) : '+', (1, 1) : '-'})
> 
>    qubit_ham = qml.parity_transform(w, n=6)

Reviewer: ```suggestion
   fermi_ham = qml.fermi.FermiWord({(0, 0) : '+', (1, 1) : '-'})

   qubit_ham = qml.fermi.parity_transform(fermi_ham, n=6)
```

  + 2j * X(0) @ Z(1)
  ```

<h4>Parity Mapping</h4>

* `parity_transform`  is added for parity mapping of a fermionic Hamiltonian.
   [(#4928)](https://github.com/PennyLaneAI/pennylane/pull/4928)
   It is now possible to transform fermionic Hamiltonian to qubit Hamiltonian through parity mapping.

   ```python

>    

Reviewer 2: ```suggestion
```

Lee J. O'Riordan,
Mudit Pandey,
Alex Preciado,
Matthew Silverman,

> Diksha Dhawan.

Reviewer 2: Hi @ddhawan11, not a blocker for merging but we go in alphabetical order based on last name, so you can move to below Isaac.

Matthew Silverman,
Jay Soni,
David Wierichs,

> Justin Woodring

Reviewer 3: Weird, I wonder why this is complaining, try: 
```suggestion
Justin Woodring.
```

* `qml.map_wires` no longer fails when mapping nested quantum tapes.
  [(#4901)](https://github.com/PennyLaneAI/pennylane/pull/4901)

> * 'parity_transform' can be used for parity mapping of a fermionic hamiltonian

Reviewer 2: ```suggestion
* `parity_transform` can be used for parity mapping of a fermionic hamiltonian
```

>    [(#4928)](https://github.com/PennyLaneAI/pennylane/pull/4928)

Reviewer 2: Minor point, but example code here is always welcome.

Author: I am not sure that I understand, lines 620 and 621 are already there in the changelog. Do you mean we should add working example for parity transform here?

Reviewer: > Do you mean we should add working example for parity transform here?

Yes, see some examples in Release 0.34.0 [changelog](https://github.com/PennyLaneAI/pennylane/blob/master/doc/releases/changelog-0.34.0.md).
